src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,57 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
+    private int size;
+    private Node<T> tail;
+
+    public MateStack() {
+        tail = new Node<>(null, null);
+    }
+
     public void push(T value) {
+        if (size == 0) {
+            tail.value = value;
+            size++;
+            return;
+        }
```

Comment:
You don't have to create a node in the constructor, then you don't need this code